pyigm/fN/tau_eff.py

```diff
@@ -0,0 +1,307 @@
+                  cumul=None, verbose=False):
+    """ tau effective from HI Lyman series absorption
+
+    Parameters
+    ----------
+    ilambda : float
+        Observed wavelength (Ang)
+    zem : float
+        Emission redshift of the source [sets which Lyman lines are included]
+    bva : float
+         -- Characteristics Doppler parameter for the Lya forest
+         -- [Options: 24, 35 km/s]
+    NHI_MIN : float
+         -- Minimum log HI column for integration [default = 11.5]
+    NHI_MAX : float
+         -- Maximum log HI column for integration [default = 22.0]
+    fNz : Boolean (False)
+         -- Inputs f(N,z) instead of f(N,X)
+    cosmo : astropy.cosmology (None)
+         -- Cosmological model to adopt (as needed)
+    cumul : List of cumulative sums
+         -- Recorded only if cumul is not None
+
+    Returns
+    -------
+    teff : float
+      Total effective opacity of all lines contributing
+
+    ToDo:
+      1. Parallelize the Lyman loop
+    """
+    # Lambda
+    if not isinstance(ilambda, float):
+        raise ValueError('tau_eff: ilambda must be a float for now')
+    Lambda = ilambda
+    if not isinstance(Lambda,u.quantity.Quantity):
+        Lambda = Lambda * u.AA # Ang
+
+    # Read in EW spline (if needed)
+    if EW_spline == None:
+        if int(bval) == 24:
+            EW_FIL = pyigm_path+'/data/fN/EW_SPLINE_b24.yml'
+            with open(EW_FIL, 'r') as infile:
+                EW_spline = yaml.load(infile)  # dict from mk_ew_lyman_spline
+        else:
+            raise ValueError('tau_eff: Not ready for this bvalue %g' % bval)
+
+    # Lines
+    HI = LineList('HI')
+    wrest = HI._data['wrest']
+
+    # Find the lines
+    gd_Lyman = wrest[(Lambda/(1+zem)) < wrest]
+    nlyman = len(gd_Lyman)
+    if nlyman == 0:
+        if verbose:
+            print('igm.tau_eff: No Lyman lines covered at this wavelength')
+        return 0
+
+    # N_HI grid
+    lgNval = NHI_MIN + (NHI_MAX-NHI_MIN)*np.arange(N_eval)/(N_eval-1) # Base 10
+    dlgN = lgNval[1]-lgNval[0]
+    Nval = 10.**lgNval
+    teff_lyman = np.zeros(nlyman)
+
+    # For cumulative
+    if not cumul is None:
+        cumul.append(lgNval)
+
+    # Loop on the lines
+    for qq, line in enumerate(gd_Lyman): # Would be great to do this in parallel...
+                             # (Can pack together and should)
+        # Redshift
+        zeval = ((Lambda / line) - 1).value
+        if zeval < 0.:
+            teff_lyman[qq] = 0.
+            continue
+        # Cosmology
+        if fNz is False:
+            if cosmo not in locals():
+                cosmo = FlatLambdaCDM(H0=70, Om0=0.3) # Vanilla
+            #dxdz = (np.fabs(xigmu.cosm_xz(zeval-0.1, cosmo=cosmo)-
+            #            xigmu.cosm_xz(zeval+0.1,cosmo=cosmo)) / 0.2 )
+            #xdb.set_trace()
+            dxdz = pyigmu.cosm_xz(zeval,cosmo=cosmo,flg_return=1)
+        else: dxdz = 1. # Code is using f(N,z)
```

Comment:
nice option, but a bit inconsistent with what's above perhaps...  It would be nice to have this option at the fn level